timesketch/api/v1/resources/sketch.py

```diff
@@ -65,28 +65,86 @@ def get(self):
         Returns:
             List of sketches (instance of flask.wrappers.Response)
         """
+        args = self.parser.parse_args()
+        scope = args.get('scope', None)
+        page = args.get('page', 1)
+        search_query = args.get('search_query', None)
+
         if current_user.admin:
             sketch_query = Sketch.query
         else:
             sketch_query = Sketch.all_with_acl()
 
-        filtered_sketches = sketch_query.filter(
+        base_filter = sketch_query.filter(
+            not_(Sketch.Status.status == 'deleted'),
+            not_(Sketch.Status.status == 'archived'),
+            Sketch.Status.parent).order_by(Sketch.updated_at.desc())
+
+        base_filter_with_archived = sketch_query.filter(
             not_(Sketch.Status.status == 'deleted'),
-            Sketch.Status.parent).order_by(Sketch.updated_at.desc()).all()
+            Sketch.Status.parent).order_by(Sketch.updated_at.desc())
 
-        # Just return a subset of the sketch objects to reduce the amount of
-        # data returned.
+        filtered_sketches = base_filter_with_archived
         sketches = []
-        for sketch in filtered_sketches:
-            sketches.append({
+        return_sketches = []
+        num_hits = 0
+
+        if scope == 'recent':
+            views = View.query.filter_by(
+                user=current_user, name='').order_by(
+                View.updated_at.desc()).limit(3)
+            sketches = [view.sketch for view in views]
+            num_hits = len(sketches)
```

Comment:
and why are we basing `recent` on just the last sketches that have saved searches added to it?

is there no other better mechanism to judge recency? (when was a timeline added, when was a story added, event labeled, event starred, comment added, etc, etc)... there are so many other `changes` to a sketch...

is this maybe better to change the API in various places to add an update to the `update` timestamp for the sketch if a change is made... that is in the view resource, we update the sketch if a new view is saved or updated... in the story api, we update the timestamp for the sketch if a story is updated/changed... etc, etc ... and then this check is just to search for last update time of sketches, top X (configurable)... that way you get a miuch more `accurate` representation of recency...

Reply:
There are more avenues yes, but not easily queried. This is not just saved searches. This is the special view that is updated anytime the user searches the sketch. So activity here means sketches that you actually did some work on (i.e. searching).

I'm about to add the suggestion that you point out here, but not in this PR :)  I'm aware that there are more actions to take into account, but this is the start of it and we can make it cover more if we think it is useful later.

Reply:
And this is the list that the user was active on, not based on what anyone else did on the sketch

Reply:
ok, but can you add a TODO here to include other activity, since as you said, this only covers one use case of "activity", that is just the fact that the user searched something in the sketch.. but there are also other ways to interact with the sketch (run an analyzer, bla bla bla)

Reply:
we might even include a "user visited the sketch" ?

Reply:
Done!